public/app/plugins/gen.go

```diff
@@ -33,6 +36,116 @@ var skipPlugins = map[string]bool{
 const sep = string(filepath.Separator)
 
 func main() {
+	if len(os.Args) > 1 {
+		log.Fatal(fmt.Errorf("plugin thema code generator does not currently accept any arguments\n, got %q", os.Args))
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(fmt.Errorf("could not get working directory: %s", err))
+	}
+	grootp := strings.Split(cwd, sep)
+	groot := filepath.Join(sep, filepath.Join(grootp[:len(grootp)-3]...))
+	lib := cuectx.GrafanaThemaRuntime()
+
+	pluginKindGen := codejen.JennyListWithNamer(func(decl *codegen.PluginTreeDeclForGen) string {
+		return decl.Tree.RootPlugin().Meta().Id
+	})
+
+	outputFile := filepath.Join("pkg", "plugins", "pfs", "corelist", "corelist_load_gen.go")
+	pluginKindGen.Append(codegen.PluginTreeListJenny(outputFile),
+		codejen.AdaptOneToMany(codegen.FlattenJenny(corecodegen.GoTypesJenny("pkg/tsdb", nil)), mapToDecls))
+
+	// composableKindsGen := codejen.JennyListWithNamer(func(decl *corecodegen.DeclForGen) string {
+	// 	return decl.Meta.Common().MachineName
+	// })
+
+	// All the jennies that comprise the composable kinds generator pipeline
+	// composableKindsGen.Append(
+	// 	corecodegen.GoTypesJenny("pkg/tsdb", nil),
+	// 	corecodegen.TSTypesJenny("public/app/plugins", &corecodegen.TSTypesGeneratorConfig{
+	// 		GenDirName: func(decl *corecodegen.DeclForGen) string {
+	// 			// FIXME this hardcodes always generating to experimental dir. OK for now, but need generator fanout
+	// 			return filepath.Join(decl.Meta.Common().MachineName, "x")
+	// 		},
+	// 	}),
+	// )
+
+	// composableKindsAdapter := AdaptTest[*corecodegen.DeclForGen](composableKindsGen, func(ptDecl *codegen.PluginTreeDeclForGen) []*corecodegen.DeclForGen {
+	// 	log.Printf("adapting plugin: %s", ptDecl.Tree.RootPlugin().Meta().Id)
+	// 	slots := ptDecl.Tree.RootPlugin().SlotImplementations()
+	// 	decls := []*corecodegen.DeclForGen{}
+
+	// 	for k, slot := range slots {
+	// 		log.Printf("  slot: %s, path: %s", slot.Name(), slot.Underlying().Path().String())
+	// 		someDecl := &kindsys.SomeDecl{
+	// 			V: slot.Underlying(),
+	// 			Meta: kindsys.ComposableMeta{
+	// 				CommonMeta: kindsys.CommonMeta{
+	// 					Name:              k,
+	// 					MachineName:       k,
+	// 					PluralName:        k + "s",
+	// 					PluralMachineName: k + "s",
+	// 				},
+	// 				CurrentVersion: slot.Latest().Version(),
```

Comment:
This should all be populated already by the loader, but i think that goes to your other comment - that loader doesn't properly exist yet, and the current form of what's in `models.cue` doesn't actually follow `#ComposableKind` spec, either